tests/tlsio_esp8266_ut/tlsio_esp8266_ut.c

```diff
@@ -0,0 +1,733 @@
+
+
+        //REGISTER_GLOBAL_MOCK_HOOK(OptionHandler_Create, my_OptionHandler_Create);
+        REGISTER_GLOBAL_MOCK_HOOK(mallocAndStrcpy_s, real_mallocAndStrcpy_s);
+        REGISTER_GLOBAL_MOCK_HOOK(lwip_select, my_lwip_select);
+        REGISTER_GLOBAL_MOCK_HOOK(SSL_write, my_SSL_write);
+        REGISTER_GLOBAL_MOCK_HOOK(SSL_CTX_free, my_SSL_CTX_free);
+        REGISTER_GLOBAL_MOCK_HOOK(SSL_free, my_SSL_free);
+        REGISTER_GLOBAL_MOCK_HOOK(SSL_read, my_SSL_read);
+        REGISTER_GLOBAL_MOCK_HOOK(SSL_connect, my_SSL_connect);
+        REGISTER_GLOBAL_MOCK_HOOK(SSL_set_fd, my_SSL_set_fd);
+        REGISTER_GLOBAL_MOCK_HOOK(SSL_new, my_SSL_new);
+        REGISTER_GLOBAL_MOCK_HOOK(SSL_set_fragment, my_SSL_set_fragment);
+        REGISTER_GLOBAL_MOCK_HOOK(SSL_CTX_new, my_SSL_CTX_new);
+        REGISTER_GLOBAL_MOCK_HOOK(netconn_gethostbyname, my_netconn_gethostbyname);
+        REGISTER_GLOBAL_MOCK_HOOK(TLSv1_client_method, my_TLSv1_client_method);
+        REGISTER_GLOBAL_MOCK_HOOK(bind, my_bind);
+        REGISTER_GLOBAL_MOCK_HOOK(connect, my_connect);
+
+        /**
+         * You can initialize other global variables here, for instance image that you have a standard void* that will be converted
+         *   any pointer that your test needs.
+         */
+        g_GenericPointer = malloc(1);
+        ASSERT_IS_NOT_NULL(g_GenericPointer);
+    }
+
+    /**
+     * The test suite will call this function to cleanup your machine.
+     * It is called only once, after all tests is done.
+     */
+    TEST_SUITE_CLEANUP(TestClassCleanup)
+    {
+        free(g_GenericPointer);
+
+        umock_c_deinit();
+
+        TEST_MUTEX_DESTROY(g_testByTest);
+        TEST_DEINITIALIZE_MEMORY_DEBUG(g_dllByDll);
+    }
+
+    /**
+     * The test suite will call this function to prepare the machine for the new test.
+     * It is called before execute each test.
+     */
+    TEST_FUNCTION_INITIALIZE(initialize)
+    {
+        if (TEST_MUTEX_ACQUIRE(g_testByTest))
+        {
+            ASSERT_FAIL("Could not acquire test serialization mutex.");
+        }
+
+        umock_c_reset_all_calls();
+    }
+
+    /**
+     * The test suite will call this function to cleanup your machine for the next test.
+     * It is called after execute each test.
+     */
+    TEST_FUNCTION_CLEANUP(cleans)
+    {
+        TEST_MUTEX_RELEASE(g_testByTest);
+    }
+
+    /* Tests_SRS_TLSIO_SSL_ESP8266_99_019: [ The tlsio_openssl_dowrok succeed]*/
+    TEST_FUNCTION(tlsio_openssl_dowork__succeed)
+    {
+        ///arrange
+        int result = 0;
+        TLS_IO_INSTANCE instance;
+        instance.on_bytes_received = on_bytes_received;
+
+        const IO_INTERFACE_DESCRIPTION* tlsioInterfaces = tlsio_openssl_get_interface_description();
+        ASSERT_IS_NOT_NULL(tlsioInterfaces);
+
+        umock_c_reset_all_calls();
+        EXPECTED_CALL(SSL_read(IGNORED_PTR_ARG, IGNORED_PTR_ARG, IGNORED_NUM_ARG));
```

Comment:
Please add an unit test to check the callback when dowork read new valid data. And another if no new data is available.